snuba/clickhouse/http.py

```diff
@@ -39,12 +39,38 @@ def __default(self, value: Any) -> Any:
         else:
             raise TypeError
 
-    def encode(self, value: WriterTableRow) -> JSONRow:
+    def encode(self, value: WriterTableRow) -> bytes:
         return cast(
             bytes, rapidjson.dumps(value, default=self.__default).encode("utf-8")
         )
 
 
+class ValuesRowEncoder(Encoder[bytes, WriterTableRow]):
+    def __init__(self, columns: Iterable[str]) -> None:
+        self.__columns = columns
+
+    def encode_value(self, value: Any) -> str:
+        if isinstance(value, str):
+            return value
+
+        if isinstance(value, int):
+            return str(value)
+
+        if isinstance(value, list):
+            for x in value:
+                if not isinstance(x, int):
+                    raise TypeError("ints should only contain arrays")
+
+            return "[" + ",".join([str(x) for x in value]) + "]"
+
+        return "0"
+
+    def encode(self, row: WriterTableRow) -> bytes:
+        ordered_columns = [self.encode_value(row[column]) for column in self.__columns]
+        ordered_columns_str = ",".join(ordered_columns)
+        return f"({ordered_columns_str})".encode("utf-8")
```

Comment:
This should be a lot more fleshed out before it's an active code path but I copied it from my other PR to make sure we had enough data flowing through (e.g. with the table schema's columns in order) for it to work